In [22]:
class serious:
    def __init__(self):
        self.name=[]
    def printy(self,john):
        self.name.append(john)

        print(self.name)

d=serious()
d.printy("jaise")


['jaise']


In [36]:
#Create a sportMart class , where you will have
#-> INVENTORY/ shelf of items
#-> Orders of customers 

#create csv file which will store your inventory details and order details 

#with the help of file handling techniques in python, read the file and 
#create an object for trinity store and populate the inventory items and orders into the trinity StopIteration

#To make sure that you have added all the items in your file, use a display method to show your inventory and order history  class sportMart:

class sportMart:
    def __init__(self):
         self.inventory = {}
         self.orders = {}
    
    def createInventory(self, productID,ProductName,Qunatity,Price):
         temp = {
             "productid":productID,
             "productname":ProductName,
          "quantity":Qunatity,
             "price": Price
         }   
         self.inventory[productID] = temp

    def createOrder(self,Orderid,ProductID,Quantity,Price,Total):
         temp = {
             "orderid":Orderid,
             "productid":ProductID,
             "quantity":Quantity,
             "price":Price,
             "total":Total
               }  
         self.orders[Orderid]= temp

    def printOrders(self):
         print(self.orders)

    def printInventory(self):
         print(self.inventory)

    def updateqnty(self):
              


trinity = sportMart()
orders = open("orders.csv","r")
o_header = orders.readline() #removes first line 
o_data =orders.readlines()
for data in o_data:
     tmp = data.strip().split(',')
     trinity.createOrder(tmp[0],tmp[1],tmp[2],tmp[3],tmp[4])

trinity.printOrders()  

trinity = sportMart()
inventory = open("inventory.csv","r")
i_header = inventory.readline()
i_data =inventory.readlines()
for data in i_data:
     tmp = data.strip().split(',')
     trinity.createInventory(tmp[0],tmp[1],tmp[2],tmp[3])

trinity.printInventory() 


{'OD0001': {'orderid': 'OD0001', 'productid': 'PID001', 'quantity': '2', 'price': '1000', 'total': '2000'}, 'OD0002': {'orderid': 'OD0002', 'productid': 'PID002', 'quantity': '5', 'price': '1500', 'total': '7500'}, 'OD0003': {'orderid': 'OD0003', 'productid': 'PID003', 'quantity': '2', 'price': '300', 'total': '600'}, 'OD0004': {'orderid': 'OD0004', 'productid': 'PID004', 'quantity': '6', 'price': '400', 'total': '2400'}, 'OD0005': {'orderid': 'OD0005', 'productid': 'PID005', 'quantity': '9', 'price': '6700', 'total': '60300'}, 'OD0006': {'orderid': 'OD0006', 'productid': 'PID006', 'quantity': '4', 'price': '3400', 'total': '13600'}, 'OD0007': {'orderid': 'OD0007', 'productid': 'PID001', 'quantity': '1', 'price': '2000', 'total': '2000'}, 'OD0008': {'orderid': 'OD0008', 'productid': 'PID002', 'quantity': '8', 'price': '5600', 'total': '44800'}, 'OD0009': {'orderid': 'OD0009', 'productid': 'PID005', 'quantity': '8', 'price': '450', 'total': '3600'}, 'OD0010': {'orderid': 'OD0010', 'prod

{'OD0001': {'orderid': 'OD0001', 'productid': 'PID001', 'quantity': '2', 'price': '1000', 'total': '2000'}, 'OD0002': {'orderid': 'OD0002', 'productid': 'PID002', 'quantity': '5', 'price': '1500', 'total': '7500'}, 'OD0003': {'orderid': 'OD0003', 'productid': 'PID003', 'quantity': '2', 'price': '300', 'total': '600'}, 'OD0004': {'orderid': 'OD0004', 'productid': 'PID004', 'quantity': '6', 'price': '400', 'total': '2400'}, 'OD0005': {'orderid': 'OD0005', 'productid': 'PID005', 'quantity': '9', 'price': '6700', 'total': '60300'}, 'OD0006': {'orderid': 'OD0006', 'productid': 'PID006', 'quantity': '4', 'price': '3400', 'total': '13600'}, 'OD0007': {'orderid': 'OD0007', 'productid': 'PID001', 'quantity': '1', 'price': '2000', 'total': '2000'}, 'OD0008': {'orderid': 'OD0008', 'productid': 'PID002', 'quantity': '8', 'price': '5600', 'total': '44800'}, 'OD0009': {'orderid': 'OD0009', 'productid': 'PID005', 'quantity': '8', 'price': '450', 'total': '3600'}, 'OD0010': {'orderid': 'OD0010', 'productid': 'PID006', 'quantity': '5', 'price': '3456', 'total': '17280'}}
{'PID001': {'productid': 'PID001', 'productname': 'Bat', 'quantity': '89', 'price': '1000'}, 'PID002': {'productid': 'PID002', 'productname': 'Ball', 'quantity': '50', 'price': '1500'}, 'PID003': {'productid': 'PID003', 'productname': 'Badminton Racket', 'quantity': '230', 'price': '300'}, 'PID004': {'productid': 'PID004', 'productname': 'Stumps', 'quantity': '1000', 'price': '400'}, 'PID005': {'productid': 'PID005', 'productname': 'Football', 'quantity': '500', 'price': '6700'}, 'PID006': {'productid': 'PID006', 'productname': 'Basket Ball', 'quantity': '300', 'price': '3400'}}


In [17]:
n="Nandhana"
for i in range(9):
    print(n)
    i=i+1
file=open("name.csv","w+")
file.write(n+"\n")
file.close()




Nandhana
Nandhana
Nandhana
Nandhana
Nandhana
Nandhana
Nandhana
Nandhana
Nandhana


In [29]:
class students:
    def _init_(self):
        self.students = {}

    def addStudents(self,id,Name,Course) :
        temp = {
            "name":Name,
            "course": Course
        }

        self.students[id]= temp

    def printing(self):
        print(self.students)
      haa = students()
userinput = int(input("Enter the No of inputs:"))
for i in range (userinput):
    id = int(input("Enter the ID: "))
    name = input("Enter the Name: ")
    course = input("Enter the Course: ")
    haa.addStudents(id,name,course)
haa.printing()


In [32]:
class sportMart:
    def __init__(self):
        self.inventory = {}
        self.orders = {}

    def createInventory(self,ProductID,ProductName,Quantity,Price):
        temp = {
            "productid":ProductID,
            "productname":ProductName,
            "quantity":Quantity,
            "price":Price
        }     
        self.inventory[ProductID]= temp

    def createOrders(self,Orderid,ProductID,Quantity,Price,Total):
        temp = {
            "orderid":Orderid,
            "productid":ProductID,
            "quantity":Quantity,
            "price":Price,
            "total":Total
        }
        self.orders[Orderid]= temp

    def printOrders(self):
        print(self.orders)

    def printInventory(self):
        print(self.inventory)    

trinity = sportMart()
orders = open("orders.csv","r+")
o_head

        
                   

<class '__main__.sportMart'>


In [35]:
class sportMart:
    def __init__(self):
        self.inventory = {}
        self.orders = {}
    
    def createInventory(self, productID,ProductName,Qunatity,Price):
        temp = {
            "productid":productID,
            "productname":ProductName,
            "quantity":Qunatity,
            "price": Price
        }   
        self.inventory[productID] = temp

    def createOrder(self,Orderid,ProductID,Quantity,Price,Total):
        temp = {
            "orderid":Orderid,
            "productid":ProductID,
            "quantity":Quantity,
            "price":Price,
            "total":Total
              }  
        self.orders[Orderid]= temp

    def printOrders(self):
        print(self.orders)

    def printInventory(self):
        print(self.inventory)


trinity = sportMart()
orders = open("orders.csv","r")
o_header = orders.readline()
o_data =orders.readlines()
for data in o_data:
    tmp = data.strip().split(',')
    trinity.createOrder(tmp[0],tmp[1],tmp[2],tmp[3],tmp[4])

trinity.printOrders()   

trinity = sportMart()
inventory = open("inventory.csv","r")
i_header = inventory.readline()
i_data =inventory.readlines()
for data in i_data:
    tmp = data.strip().split(',')
    trinity.createInventory(tmp[0],tmp[1],tmp[2],tmp[3])

trinity.printInventory()  

{'OD0001': {'orderid': 'OD0001', 'productid': 'PID001', 'quantity': '2', 'price': '1000', 'total': '2000'}, 'OD0002': {'orderid': 'OD0002', 'productid': 'PID002', 'quantity': '5', 'price': '1500', 'total': '7500'}, 'OD0003': {'orderid': 'OD0003', 'productid': 'PID003', 'quantity': '2', 'price': '300', 'total': '600'}, 'OD0004': {'orderid': 'OD0004', 'productid': 'PID004', 'quantity': '6', 'price': '400', 'total': '2400'}, 'OD0005': {'orderid': 'OD0005', 'productid': 'PID005', 'quantity': '9', 'price': '6700', 'total': '60300'}, 'OD0006': {'orderid': 'OD0006', 'productid': 'PID006', 'quantity': '4', 'price': '3400', 'total': '13600'}, 'OD0007': {'orderid': 'OD0007', 'productid': 'PID001', 'quantity': '1', 'price': '2000', 'total': '2000'}, 'OD0008': {'orderid': 'OD0008', 'productid': 'PID002', 'quantity': '8', 'price': '5600', 'total': '44800'}, 'OD0009': {'orderid': 'OD0009', 'productid': 'PID005', 'quantity': '8', 'price': '450', 'total': '3600'}, 'OD0010': {'orderid': 'OD0010', 'prod

In [37]:
class sportmart:
    def _init_(self):
        self.inventory={}
        self.orders={}

    def  createorder(self,Orderid,ItemName,Quantity,Price,Total):
        temp={"orderid":Orderid,
              "itemname":ItemName,
              "quantity":Quantity,
              "price":Price,
              "total":Total
              }
        
        self.orders[Orderid]=temp
        return temp

    def createinventory(self,ProductID,ProductName,Quantity,Price):
        temp={"productid":ProductID,
             "productname":ProductName,
             "quantity":Quantity,
             "price":Price
             }
        self.inventory[ProductID]=temp
    
    def printorders(self):
        print("\n",self.orders)

    def printinventory(self):
        print(self.inventory)

    def generateid(self):
        first="OD"
        orderid=''
        for i in range(1,4,1):
            orderid+=str(random.randint(i,9))
        orderid=first+orderid
        return orderid
    
    def updateinventory(self,name,qty):
        for item in self.inventory.values():
            if item['productname']==name:
                item['quantity']=int(item['quantity'])-qty
        print("\nThe updated inventory data:\n")
        self.printinventory()

    def storefile(self):
        file=open("newdata.csv","a+")
        file.write(str(self.inventory))
        file.close()

trinity=sportmart()


while True:
    print("\nMenu Driven Program\n")
    print("1. Print Orders\n2. Print Inventory\n3. Export Data")
    choice=int(input("Enter your choice:"))
    if choice==1:
        orders=open("Order.csv","r")
        orders_header=orders.readline()
        orders_orders=orders.readlines()
        for item in orders_orders:
            temp=item.strip().split(',')
            trinity.createorder(temp[0],temp[1],temp[2],temp[3],temp[4])
        trinity.printorders()
    
    elif choice==2:
        inventory=open("Inventory.csv","r")
        inventory_header=inventory.readline()
        inventory_list=inventory.readlines()
        for item in inventory_list:
            temp=item.strip().split(',')
            trinity.createinventory(temp[0],temp[1],temp[2],temp[3])
        trinity.printinventory()

    elif choice==3:
        add={}
        itemname=input("Enter item name:").title()
        quantity=int(input("Enter quantity:"))
        price=int(input("Enter price:"))
        total=price*quantity
        id=trinity.generateid()
        #add=trinity.createorder(id,itemname,quantity,price,total)
        trinity.orders[id] = trinity.createorder(id,itemname,quantity,price,total)
        trinity.printorders()
        print("\n\n")
        trinity.updateinventory(itemname,quantity)
    elif choice==4:
        trinity.storefile()
    elif choice==5:
        exit()
    else:
        print("Wrong choice")


Menu Driven Program

1. Print Orders
2. Print Inventory
3. Export Data


FileNotFoundError: [Errno 2] No such file or directory: 'Order.csv'